pkg/resources/table_constraint.go

```diff
@@ -307,26 +356,24 @@ func ReadTableConstraint(_ *schema.ResourceData, _ interface{}) error {
 // UpdateTableConstraint implements schema.UpdateFunc.
 func UpdateTableConstraint(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+
 	tc := tableConstraintID{}
 	tc.parse(d.Id())
-	formattedTableID := snowflakeValidation.ParseAndFormatFullyQualifiedObectID(tc.tableID)
-
-	builder := snowflake.NewTableConstraintBuilder(tc.name, tc.constraintType, formattedTableID)
 
-	/* "unsupported feature comment error message"
-	if d.HasChange("comment") {
-		_, new := d.GetChange("comment")
-		_, err := db.Exec(builder.SetComment(new.(string)))
-		if err != nil {
-			return fmt.Errorf("error setting comment for table constraint %v", tc.name)
-		}
-	}*/
+	tableIdentifier, err := getTableIdentifier(tc.tableID)
+	if err != nil {
+		return err
+	}
 
 	if d.HasChange("name") {
 		_, n := d.GetChange("name")
-		_, err := db.Exec(builder.Rename(n.(string)))
+		constraintRequest := sdk.NewTableConstraintRenameActionRequest().WithOldName(tc.name).WithNewName(n.(string))
+		alterStatement := sdk.NewAlterTableRequest(*tableIdentifier).WithConstraintAction(sdk.NewTableConstraintActionRequest().WithRename(constraintRequest))
+		err = client.Tables.Alter(ctx, alterStatement)
 		if err != nil {
-			return fmt.Errorf("error renaming table constraint %v err = %w", tc.name, err)
+			return fmt.Errorf("error renaming table constraint %v err = %w", &tc.name, err)
```

Comment:
`&tc.name` will print a pointer, right ? If so, remove `&`

Reply:
True. Will change in the following change then.